scripts/setup/sha256-file-to

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+if [ "$#" -ne 3 ]; then
+    echo "Usage:"
+    echo "  sha256-file-to SHA256 http://FETCH/FROM DST"
+    echo
+    echo "SHA256 is the sha256sum of the file fetched; the file is"
+    echo "placed in DST."
+    exit 1
+fi
+
+set -e
+set -x
+
+SHA256="$1"
+URL="$2"
+DST="$3"
+
+# Work in a tmpdir which we clean up at the end
+tmpdir="$(mktemp -d)"
+trap 'rm -r "$tmpdir"' EXIT
+cd "$tmpdir"
+
+# Fetch to a predictable name, not whatever curl guesses from the URL
+LOCALFILE="output"
+curl -fL --retry 3 -o "$LOCALFILE" "$URL"
```

Comment:
I know in the provisioner I've done some changes to ensure *all* steps get curl args plumbed around (since proxies and the like are sometimes necessary, particularly to GitHub) and only a few did such plumbing. Do you think it's worth adding that extra bit of plumbing here as well, or (1) will this not be run outside of ZC/CZO contexts anyway, or (2) are we just going to lean on `HTTP_PROXY` in env and assume no other `CURL_OPTIONS` would be useful here? (or 3: something I didn't list, this is by no means an exhaustive set of our options 😄)

Reply:
1. I think it's possible other deploys will use this integration, so making sure it works with proxy settings is a good point
2. We pull a lot of things from the internet in production installs and upgrades, and we expect that `HTTPS_PROXY` will plumb through for them to work.  I've never tested this myself, but I believe that folks do so successfully. 
 If that's not being sufficient, we'll need to fix more than this location. 

   I think that 0acb96ea9debfb706fd5897aa9f84e10d38cda3e was about plumbing the env vars through the `sudo` in dev, and I believe we don't use `sudo` in the prod install/upgrade paths at all, so I don't think it has a direct equivalent here.